### PR TITLE
Ensure Travis is using npm5 for builds

### PR DESCRIPTION
**Pull request by gemmaleigh**
We can't be sure that npm5 is installed globally when dependencies are installed - in which case the package-lock.json file won't be updated. 

Use this check to ensure that if dependencies are added to
package.json, that the package-lock.json file has also been updated.

Also ensure npm5 is installed globally for Travis.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,17 +1,21 @@
 language: node_js
 node_js:
-- 6.0
+- 7
 notifications:
   email: false
 sudo: false
 before_install:
-  openssl aes-256-cbc -K $encrypted_ca4a154c2e6b_key -iv $encrypted_ca4a154c2e6b_iv -in config/tenon.json.enc -out config/tenon.json -d
+  - if [[ `npm -v` != 5* ]]; then npm i -g npm@5.0.3; fi
```
Comment (tvararu) [inline .travis.yml]: Should we try `npm i -g npm@latest`? If this does eventually end up biting us we can lock to a particular version.

File: check-package-lock.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# Check to see whether the package-lock.json file needs updating.
+
+echo "checking package-lock.json for changes"
+
+if git diff --exit-code --quiet package-lock.json; then
+  echo "no changes have been made to package-lock.json"
+else
+  echo "package-lock.json needs updating, run npm install and commit"
```
Comment (tvararu) [inline check-package-lock.sh]: You need to run it on a new version of npm, so maybe this is better:

```
echo "package-lock.json needs updating, run npm install on npm@5+ and commit"
```

Comment (gemmaleigh) [inline check-package-lock.sh]: good point @tvararu, will update!

File: .travis.yml [hunk 3]
```diff
@@ -1,17 +1,21 @@
 language: node_js
 node_js:
-- 6.0
+- 7
 notifications:
   email: false
 sudo: false
 before_install:
-  openssl aes-256-cbc -K $encrypted_ca4a154c2e6b_key -iv $encrypted_ca4a154c2e6b_iv -in config/tenon.json.enc -out config/tenon.json -d
+  - if [[ `npm -v` != 5* ]]; then npm i -g npm@latest; fi
```
Comment (tvararu) [inline .travis.yml]: Sorry, didn't read the conditional when I suggested the `@latest` earlier; to make sure we don't reinstall `@latest` constantly when `npm@6` comes out we can do:

```bash
- if [[ `npm -v | cut -d. -f1` -lt 5 ]]; then npm i -g npm@latest; fi
```

`cut` will split the version number by the `.` delimeter and then give us the first field, which is the major number in the version. If this is less than 5, install the latest version of `npm`.

**Comment by gemmaleigh** [comment]
This one needs merging first, then the package-lock.json file will need updating for open PRs.

**Pull request merged**